### PR TITLE
add time filter on asset events

### PR DESCRIPTION
**Pull request by nishant-gupta-sh**
---
Adding timestamp filter on asset events similar to this PR: https://github.com/apache/airflow/pull/44654

### CODE REVIEW

File: airflow/api_fastapi/core_api/routes/public/assets.py
Comment (pierrejeambrun) [inline airflow/api_fastapi/core_api/routes/public/assets.py:198]: We also need tests for `api_fastapi` endpoint.

**Review by pierrejeambrun — COMMENTED** [review]
Nice. 

Need tests for the airflow 3 endpoint.

File: airflow/api_fastapi/core_api/routes/public/assets.py
Comment (nishant-gupta-sh) [reply airflow/api_fastapi/core_api/routes/public/assets.py:152]: Are there currently tests for any api_fastapi endpoints?

Comment (nishant-gupta-sh) [reply airflow/api_fastapi/core_api/routes/public/assets.py:198]: Tests added

File: tests/api_fastapi/core_api/routes/public/test_assets.py
Comment (pierrejeambrun) [inline tests/api_fastapi/core_api/routes/public/test_assets.py]: If we always create 3 event, no need to parametrize it.

Also `expected_ids` should be at the end (it's the final assertion), while `3` (the number of event we create should be at the beginning because it's part of the initial `Arrange` phase. (timestamp_gte etc... should be in the middle because it's the `Act` part of the test)

**Review by pierrejeambrun — APPROVED** [review]
Looking good.

1 minor nit, can merge after.

**Review by pierrejeambrun — COMMENTED** [review]
As mentioned in https://github.com/apache/airflow/pull/44654#issuecomment-2538217648 we cannot add feature to the legacy API, can we remove the part where we update the legacy (airflow 2.x) API, and only leave the airflow 3 update. (`api_fastapi`, etc.)

Then I think we can merge.

**Review by pierrejeambrun — APPROVED** [review]
Nice thanks.

**Comment by pierrejeambrun** [comment]
CI needs fixing before merging.

**Review by pierrejeambrun — APPROVED** [review]

**Comment by pierrejeambrun** [comment]
I would recommend installing the pre-commit hooks and running them locally so you don't have to wait for the CI to know that some formatting is wrong.

You can take a look at https://github.com/apache/airflow/blob/main/contributing-docs/08_static_code_checks.rst to get more information

**Review by jscheffl — APPROVED** [review]

**Pull request merged**
